src/internal/operations/backup.go

```diff
@@ -140,18 +141,19 @@ func (op *BackupOperation) Run(ctx context.Context) (err error) {
 		observe.Complete()
 	}()
 
-	ctx, flushMetrics := events.NewMetrics(ctx, logger.Writer{Ctx: ctx})
-	defer flushMetrics()
+	var runnable bool
 
-	//-----
-	// Precheck
-	//-----
-	err = Precheck(ctx, op.account, op.Selectors.PathService(), op.Selectors.DiscreteOwner)
-	if err != nil {
+	// IsBackupRunnable checks if the user has services enabled to run a backup.
+	// it also checks for conditions like mailbox full.
+	runnable, op.canMakeDeltaQueries, err = op.bp.IsBackupRunnable(ctx, op.Selectors.PathService(), op.ResourceOwner.ID())
+	if err != nil && !runnable {
 		logger.CtxErr(ctx, err).Error("running backup")
 		op.Errors.Fail(clues.Wrap(err, "running backup"))
 	}
 
+	ctx, flushMetrics := events.NewMetrics(ctx, logger.Writer{Ctx: ctx})
```

Comment:
This should be established at the top of the func.  Is there a particular reason it was moved below the runnable check?

Reply:
It was done with intention that we initialise the metrics only if we will be going forward with the backup. This saves initialisation time.

Reply:
Initializing metrics is lightweight.  we don't need to avoid all initializations.  our core concern should be avoiding unnecessary kopia manifest/snapshot loads.  Since IsBackupRunnable does make graph api calls, we absolutely want metrics to be tracked for those calls.

Reply:
@vkamra and I discussed this the other day that it would be better to move the code up. 
We can go with whatever the best fit is here.

Reply:
As of now I have moved the metrics initialisation above to capture  `IsBackupRunnable` graph api calls.